src/sqlfluff/rules/L003.py

```diff
@@ -153,12 +144,7 @@ def _process_raw_stack(
                 # newline that ended the *current* line was in templated space.
                 # Reason: We want to ignore indentation of lines that are not
                 # present in the raw (pre-templated) code.
-                templated_line = bool(
```

Comment:
Looking up raw slices is a bit slow (as currently implemented, it scans a list of *all* the raw slices). Since this is executed for _every_ newline, it's worthwhile to make it as fast as practical.